Library/Homebrew/readall.rb

```diff
@@ -66,19 +66,21 @@ def valid_casks?(_casks, os_name: nil, arch: nil)
       true
     end
 
-    def valid_tap?(tap, options = {})
+    def valid_tap?(tap, aliases: false, no_simulate: false, os_arch_combinations: nil)
       success = true
 
-      if options[:aliases]
+      if aliases
         valid_aliases = valid_aliases?(tap.alias_dir, tap.formula_dir)
         success = false unless valid_aliases
       end
-      if options[:no_simulate]
+      if no_simulate
         success = false unless valid_formulae?(tap.formula_files)
         success = false unless valid_casks?(tap.cask_files)
       else
-        os_names = [*MacOSVersion::SYMBOLS.keys, :linux]
-        os_names.product(OnSystem::ARCH_OPTIONS).each do |os, arch|
+        # TODO: Remove this default case once `--os` and `--arch` are passed explicitly to `brew readall` in CI.
+        os_arch_combinations ||= [*MacOSVersion::SYMBOLS.keys, :linux].product(OnSystem::ARCH_OPTIONS)
```

Comment:
Can we really remove this default case? This function is also used over in the `tap` class.

https://github.com/Homebrew/brew/blob/c265d3d330adb2707502e61cc93864fb9353a7bc/Library/Homebrew/tap.rb#L316

Reply:
I think we can. Realistically, the tap check doesn't need to simulate since as long as the tap works on the system that's tapping it, there is no problem for non-developers.

Reply:
I agree with that. The only thing I worry about is that this comment implies that the code will run fine if this line is removed without any other changes. As far as I can tell, `Readall.valid_tap?` will fail when called in `Tap` because `os_arch_combinations` will default to nil yet be used later on like it's not nil.

Reply:
Obviously you have to take context into account when removing the default case. I don't think anyone resolving this TODO would just do what it states verbatim.

Reply:
Maybe instead of a todo here you could just have a follow up PR ready before hand since it seems like a small change.

Reply:
I will probably create a follow-up PR soon anyways, but will leave the TODO here just in case.